src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/OverlayListPositioner.js

```diff
@@ -0,0 +1,123 @@
+
+const MIN_HEIGHT = 100;
+const INDENT_FROM_ANCHOR = 5;
+const PADDING_TO_WINDOW = 10;
+
+type VerticalCrop = {
+    dimensions: OverlayListDimensions,
+    touchesTopBorder: boolean,
+    touchesBottomBorder: boolean,
+};
+
+/**
+ * The class is responsible for calculating the position of the overlay-list on the screen when opened.
+ * Generally, it positions the list above an anchor element. Moreover the list is shifted, such that
+ * a designated element in the list is right above the anchor element.
+ * It is made sure that the list does not overflow the borders of the screen and even keeps a padding to the borders.
+ *
+ * The behaviour described to this point, however, is broken at one point: When the height of the list undercuts
+ * a certain minimum height, the list is positioned from the anchor element upwards or downwards depending on if
+ * the list overflows the bottom or top border of the screen.
+ */
+export default class OverlayListPositioner {
+    listHeight: number;
+    listWidth: number;
+    centeredChildRelativeTop: number;
+
+    anchorTop: number;
+    anchorLeft: number;
+    anchorWidth: number;
+    anchorHeight: number;
+
+    windowWidth: number;
+    windowHeight: number;
+
+    constructor(
+        listHeight: number,
+        listWidth: number,
+        centeredChildRelativeTop: number,
+        anchorTop: number,
+        anchorLeft: number,
+        anchorWidth: number,
+        anchorHeight: number,
+        windowWidth: number = window.innerWidth,
+        windowHeight: number = window.innerHeight,
+    ) {
+        this.listHeight = listHeight;
+        this.listWidth = listWidth;
+        this.centeredChildRelativeTop = centeredChildRelativeTop;
+        this.anchorTop = anchorTop;
+        this.anchorLeft = anchorLeft;
+        this.anchorWidth = anchorWidth;
+        this.anchorHeight = anchorHeight;
+        this.windowWidth = windowWidth;
+        this.windowHeight = windowHeight;
+    }
+
+    static dimensionsToStyle(dimensions: OverlayListDimensions): OverlayListStyle {
+        return {
+            top: dimensions.top + 'px',
+            height: dimensions.height + 'px',
+            left: dimensions.left + 'px',
+        };
+    }
+
+    getCroppedDimensions(): OverlayListDimensions {
+        // First, the list is positioned without taking the screen borders or the minimum height into account.
+        let dimensions = {
+            top: this.anchorTop - this.centeredChildRelativeTop,
+            left: this.anchorLeft - (this.listWidth - this.anchorWidth) - INDENT_FROM_ANCHOR,
+            height: this.listHeight,
+            scrollTop: 0,
+        };
+        let crop = this.cropVerticalDimensions(dimensions);
+        // If after making sure, the list does not overflow the top and the bottom border of the screen, the
+        // list succeeds the minimum height, no more steps have to be taken.
+        if (crop.dimensions.height >= MIN_HEIGHT) {
```

Comment:
It comes from the designers. A select with only two options will almost certainly never overflow the screen on either the top or bottom border. However if it does it won't get cropped any smaller because it's not bigger than 100px.